tests/test_autodiscover.cpp

```diff
@@ -64,7 +64,7 @@ TEST_F(AutodiscoverTest, EmptyAddressThrows)
     EXPECT_THROW(
         {
             auto result = ews::get_exchange_web_services_url<http_request_mock>(
-                "", ews::autodiscover_protocol::internal, credentials());
+                "", credentials());
         },
```

Comment:
As for the tests: I see that you changed the implementations so that they compile and run again. However, you also added new functionality in this PR (the URL hinting part). Where do you test this?